Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -526,6 +527,9 @@ class SourceGenerator {
                 sourcePath = path
                 sourceReference = fileReference
                 rootGroups.insert(fileReference)
+            } else if hasCustomParent {
+                sourcePath = path
```

Comment:
If the path is something like `File.file` then the parent would be the base path, and then it would fall into the first if and not this one. Maybe `hasCustomParent` should be checked for first?

Reply:
I'm looking into it!

Reply:
Fixed!